src/atlas_search.rs

```diff
@@ -32,65 +33,180 @@ use crate::bson::{doc, Bson, Document};
-    pub fn on_index(self, index: impl AsRef<str>) -> Document {
-        doc! {
-            "$search": {
-                "index": index.as_ref(),
-                self.name: self.stage,
-            }
-        }
+/// Finalize a search operator as a pending `$search` aggregation stage, allowing
+/// options to be set.
+pub fn search<T>(op: SearchOperator<T>) -> AtlasSearch {
+    AtlasSearch {
+        stage: doc! { op.name: op.spec },
+    }
+}
+
+/// A pending `$search` aggregation stage.
+pub struct AtlasSearch {
+    stage: Document,
+}
+
+impl AtlasSearch {
+    /// Parallelize search across segments on dedicated search nodes.
+    pub fn concurrent(mut self, value: bool) -> Self {
+        self.stage.insert("concurrent", value);
+        self
+    }
+
+    /// Document that specifies the count options for retrieving a count of the results.
+    pub fn count(mut self, value: Document) -> Self {
+        self.stage.insert("count", value);
+        self
+    }
+
+    /// Document that specifies the highlighting options for displaying search terms in their
+    /// original context.
+    pub fn highlight(mut self, value: Document) -> Self {
+        self.stage.insert("highlight", value);
+        self
+    }
+
+    /// Name of the Atlas Search index to use.
+    pub fn index(mut self, value: impl Into<String>) -> Self {
+        self.stage.insert("index", value.into());
+        self
+    }
+
+    /// Flag that specifies whether to perform a full document lookup on the backend database or
+    /// return only stored source fields directly from Atlas Search.
+    pub fn return_stored_source(mut self, value: bool) -> Self {
+        self.stage.insert("returnStoredSource", value);
+        self
+    }
+
+    /// Reference point for retrieving results.
+    pub fn search_after(mut self, value: impl Into<String>) -> Self {
+        self.stage.insert("searchAfter", value.into());
+        self
+    }
+
+    /// Reference point for retrieving results.
+    pub fn search_before(mut self, value: impl Into<String>) -> Self {
+        self.stage.insert("searchBefore", value.into());
+        self
+    }
+
+    /// Flag that specifies whether to retrieve a detailed breakdown of the score for the documents
+    /// in the results.
+    pub fn score_details(mut self, value: bool) -> Self {
+        self.stage.insert("scoreDetails", value);
+        self
+    }
+
+    /// Document that specifies the fields to sort the Atlas Search results by in ascending or
+    /// descending order.
+    pub fn sort(mut self, value: Document) -> Self {
+        self.stage.insert("sort", value);
+        self
+    }
+
+    /// Document that specifies the tracking option to retrieve analytics information on the search
+    /// terms.
+    pub fn tracking(mut self, value: Document) -> Self {
+        self.stage.insert("tracking", value);
+        self
+    }
+
+    /// Convert to an aggregation stage document.
+    pub fn stage(self) -> Document {
```

Comment:
I know it's more verbose, but I would prefer `into_stage` here to make it more clear that this is the final thing that needs to happen when constructing a search stage; otherwise this reads more as another constructor method